packages/utils/interactive-hooks/src/useKeyProps.ts

```diff
@@ -23,38 +29,106 @@ const noOp = () => {};
+      validKeysUp: keys,
+    },
+    windows: {
+      onKeyUp: getKeyCallbackWorker(userCallback, ...keys),
+      keyUpEvents: keys.map((keyCode) => {
+        return { key: keyCode };
+      }),
+    },
+    // win32
+    default: {
+      onKeyUp: getKeyCallbackWorker(userCallback, ...keys),
+      keyUpEvents: keys.map((keyCode) => {
+        return { key: keyCode };
+      }),
+    },
+  });
+  return keyboardProps;
+}
+
+function getKeyDownPropsWorker(userCallback: KeyCallback, ...keys: string[]): KeyPressProps {
+  const keyboardProps = Platform.select({
+    ios: undefined,
+    android: undefined,
+    macos: {
+      onKeyDown: getKeyCallbackWorker(userCallback, ...keys),
+      validKeysDown: keys,
+    },
+    windows: {
+      onKeyDown: getKeyCallbackWorker(userCallback, ...keys),
+      keyDownEvents: keys.map((keyCode) => {
+        return { key: keyCode };
+      }),
+    },
+    // win32
+    default: {
+      onKeyDown: getKeyCallbackWorker(userCallback, ...keys),
+      keyDownEvents: keys.map((keyCode) => {
+        return { key: keyCode };
+      }),
+    },
+  });
+  return keyboardProps;
+}
 
 /**
- * Re-usable hook for an onKeyUp event. noOp on unsupported platforms.
+ * Re-usable hook for an onKeyUp event.
  * @param userCallback The function you want to be called once the key has been activated on key up
  * @param keys A string of the key you want to perform some action on. If undefined, always invokes userCallback
- * @returns KeyPressProps: An object containing the correct platform specific props to  handle key press
+ * @returns KeyPressProps: An object containing the correct platform specific props to handle key press
  */
-export const useKeyUpProps = noOp2;
+export const useKeyUpProps = memoize(getKeyUpPropsWorker);
 
 /**
- * Re-usable hook for an onKeyDown event. noOp on unsupported platforms.
+ * Re-usable hook for an onKeyDown event.
  * @param userCallback The function you want to be called once the key has been activated on key down
  * @param keys A string of the key you want to perform some action on. If undefined, always invokes userCallback
  * @returns KeyPressProps: An object containing the correct platform specific props to  handle key press
  */
-export const useKeyDownProps = noOp2;
+export const useKeyDownProps = memoize(getKeyDownPropsWorker);
 
 /**
  * Re-usable hook for keyboard events. on macOS, this is onKeyDown, while on windows this is onKeyUp.
  * @param userCallback The function you want to be called once the key has been activated on key down
  * @param keys A string of the key you want to perform some action on. If undefined, always invokes userCallback
- * @returns KeyPressProps: An object containing the correct platform specific props to  handle key press
+ * @returns KeyPressProps: An object containing the correct platform specific props to handle key press
  */
-export const useKeyProps = noOp2;
+export const useKeyProps = memoize(getKeyUpPropsWorker);
```

Comment:
Doesn't seem like this is following "on macOS, this is onKeyDown, while on windows this is onKeyUp." Is that intentional?

Reply:
Good catch, fixed. I guess the reason this didn't fail snapshots is none of our snapshots with keyboard events are using macOS as their platform (iirc, they're all win32).